R/indexing.R

```diff
@@ -322,3 +357,303 @@ BasicIndexer <- R6::R6Class("BasicIndexer",
     }
   )
 )
+
+# Reference: https://github.com/zarr-developers/zarr-python/blob/5dd4a0e6cdc04c6413e14f57f61d389972ea937c/zarr/indexing.py#L585
+
+#' The Zarr OrthogonalIndexer class.
+#' @title OrthogonalIndexer Class
+#' @docType class
+#' @description
+#'  An indexer class to normalize a selection of an array and provide an iterator 
+#'  of indexes over the dimensions of an array.
+#' @param selection selection as with ZarrArray, scalar, string, or Slice. "..." and ":" supported for string 
+#' @param array ZarrArray object that will be indexed
+#' @rdname OrthogonalIndexer
+#' @keywords internal
+OrthogonalIndexer <- R6::R6Class("OrthogonalIndexer",
```

Comment:
Would you be able to add a test for this `OrthogonalIndexer` class? I know python may not have any analogous tests but maybe the current behavior can be tested like for `BasicIndexer` https://github.com/keller-mark/pizzarr/blob/main/tests/testthat/test-indexing-basic.R#L26 to prevent regressions

Reply:
Here you go 6f5d6c276f43d2ee63bc9721eb866f8eb86971ca, organized the tests a little too